main.ts

```diff
@@ -99,25 +99,28 @@ function removeCache(messages_only_content: string[]) {
 
 async function fetchFull(chat: Chat, messages: Messages) {
   let message: Response | undefined;
-  let text: string | undefined;
+  let text: string = "";
   let messageData: MessageData | undefined;
 
   for (let i = 0; i < messages.length; i += 2) {
-    text = "";
-
-    const content = messages[i]["content"];
+    const content = messages[i].content;
     message = await chat.fetch(content);
 
     const stream = events(message as Response);
     for await (const event of stream) {
-      if (!event.data) {
-        break;
+      if (!event.data || event.data === "[DONE]") {
```

Comment:
This change is not necessary. Because [DONE] is the last `event.data`, the `message` of the penultimate `event.data` is empty and the loop exits when `if (messageData["message"] == undefined)`.